src/components/field/_macro.njk

```diff
@@ -1,25 +1,25 @@
 {% from "components/error/_macro.njk" import onsError %}
 
 {% macro onsField(params) %}
-    {% if not params %}
+    {% if params is not defined %}
         {% set params = {} %}
     {% endif %}
 
     {% set field %}
-        {% if params.dontWrap %}
+        {% if params.dontWrap is defined %}
             {{ caller() }}
         {% else %}
             <p
-                {% if params.id %}id="{{ params.id }}"{% endif %}
-                class="field{% if params.classes %} {{ params.classes }}{% endif %}"
-                {% if params.attributes %}{% for attribute, value in (params.attributes.items() if params.attributes is mapping and params.attributes.items else params.attributes) %}{{ attribute }}{% if value %}="{{ value }}"{% endif %} {% endfor %}{% endif %}
+                {% if params.id is defined %}id="{{ params.id }}"{% endif %}
+                class="field{% if params.classes is defined %} {{ params.classes }}{% endif %}"
+                {% if params.attributes is defined %}{% for attribute, value in (params.attributes.items() if params.attributes is mapping and params.attributes.items else params.attributes) %}{{ attribute }}{% if value is defined %}="{{ value }}"{% endif %} {% endfor %}{% endif %}
             >
                 {{ caller() }}
             </p>
         {% endif %}
     {% endset %}
 
-    {% if params.error %}
+    {% if params.error and params.error is defined %}
```

Comment:
Duplicate param?
```suggestion
    {% if params.error is defined %}
```

Reply:
Nah I intentionally did that, its because "is defined" tests if its defined and "if params.error" just tests if its truthy

Reply:
After having a chat with Matt, maybe this isn't needed

Reply:
Remembered why I put these in now. Tests fail without them haha. So I've put them back in